### PR TITLE
feat: python 3.11 support [APE-902]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: .github/workflows/commitlint.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
         - name: Setup Python
           uses: actions/setup-python@v4
           with:
-              python-version: 3.8
+              python-version: 3.9
```
Comment (fubuloubu) [inline .github/workflows/commitlint.yaml:19]: Don't necessarily have to change these

Comment (fubuloubu) [reply .github/workflows/commitlint.yaml:19]: But, 3.10+ is much faster for running this stuff

**Review by fubuloubu — DISMISSED** [review]

**Comment by fubuloubu** [comment]
pysha3 needed for support on testing deps?

Maybe can release this with 3.11 support and just not test it until pysha3 supports

**Comment by antazoey** [comment]
> Maybe can release this with 3.11 support and just not test it until pysha3 supports

pysha3 is deprecated now, but maybe there is a different backend we could use?

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**